### PR TITLE
Add release notes for v1.4.0

### PR DESCRIPTION
**Pull request by pebrc**
Auto-generated part of the release documentation.  Preview: https://cloud-on-k8s_4116.docs-preview.app.elstc.co/guide/en/cloud-on-k8s/master/release-notes-1.4.0.html

### CODE REVIEW

**Review by charith-elastic — APPROVED** [review]
LGTM

File: docs/release-notes/1.4.0.asciidoc [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+* Remove unnecessary metrics from _nodes ES API request {pull}3860[#3860] (issue: {issue}3249[#3249])
+* Allow automatic Elasticsearch nodes discovery {pull}3837[#3837] (issues: {issue}3182[#3182], {issue}3723[#3723])
+* Enable webhook dry run on supported k8s versions {pull}3834[#3834] (issue: {issue}1233[#1233])
+* Create one transport certificates Secret per StatefulSet {pull}3828[#3828] (issue: {issue}3734[#3734])
+
+[[bug-1.4.0]]
+[float]
+=== Bug fixes
+
+* Update client-go to 0.18.14 {pull}4097[#4097] (issue: {issue}4087[#4087])
```
Comment (david-kow) [inline docs/release-notes/1.4.0.asciidoc:34]: Is that important for our release notes?

**Review by david-kow — APPROVED** [review]
LGTM, one nit

File: docs/release-notes/1.4.0.asciidoc (hunk 1 above)
Comment (pebrc) [reply docs/release-notes/1.4.0.asciidoc:34]: I left it in because it was caused by a bug report from a user. But I can remove it.

Comment (pebrc) [reply docs/release-notes/1.4.0.asciidoc:34]: I think I am gonna leave it in for the reasons stated.

**Review by david-kow — APPROVED** [review]

**Pull request merged**